query-node/mappings/storage/index.ts

```diff
@@ -227,13 +230,63 @@ export async function storage_DataObjectsUploaded({ event, store }: EventContext
 
 export async function storage_PendingDataObjectsAccepted({ event, store }: EventContext & StoreContext): Promise<void> {
   const [, , bagId, dataObjectIds] = new Storage.PendingDataObjectsAcceptedEvent(event).params
-  const dataObjects = await getDataObjectsInBag(store, bagId, dataObjectIds)
+  const dataObjects = await getDataObjectsInBag(store, bagId, dataObjectIds, ['videoThumbnail', 'videoMedia'])
+
+  // get ids of videos that are in relation with accepted data objects
+  const notUniqueVideoIds = dataObjects
+    .map(item => [item.videoMedia?.id.toString(), item.videoThumbnail?.id.toString()])
+    .flat()
+    .filter(item => item)
+  const videoIds = [...new Set(notUniqueVideoIds)]
+
+  // load videos
+  const videosPre = await store.getMany(Video, {
+    where: { id: In(videoIds) },
+    relations: videoRelationsForCounters,
+  })
+
+  // remember if videos are fully active before data objects update
+  const initialActiveStates = videosPre
+    .map(video => getVideoActiveStatus(video))
+    .filter(item => item)
+
+  // accept storage data objects
   await Promise.all(
-    dataObjects.map(async (dataObject) => {
+    dataObjects.map(async (dataObject, index) => {
       dataObject.isAccepted = true
       await store.save<StorageDataObject>(dataObject)
     })
   )
+
+  /*
+    This approach of reloading videos one by one is not optimal, but it is straightforward algorithm.
+
+    This reduces otherwise complex situation caused by `store.get*` functions not return objects
+    shared by mutliple entities (at least now). Because of that when updating for example
+    `dataObject.videoThumnail.channel.activeVideoCounter` on dataObject A, this change is not
+    reflected on `dataObject.videoMedia.channel.activeVideoCounter` on dataObject B.
+
+    We can upgrade this algorithm in the future if this event mapping proves to have serious
+    performance issues. In that case, a unit test for this mapping will be required.
+  */
+  // load relevant videos one by one and update related active-video-counters
+  await initialActiveStates.reduce(async (accPromise, initialActiveState) => {
+    await accPromise
+
+    // load refreshed version of videos and related entities (channel, channel category, category)
+
+    const video = await store.get(Video, {
```

Comment:
Because if we load entities via `getMany` in the beginning, the changes that will be made into their related entities (channels, etc.) will not be reflected. 

That's because the current implementation of Hydra seems to return duplicate/non-identical objects. So if we have `VideoA` that belongs to `ChannelA` and `VideoB` that also belongs to `ChannelB`, if we use `getMany` with `relations: ['channel']` parameter, we will receive two different `channelB` objects. If we update `activeVideoCounters` on one of them, the second one will be unaffected.

This algorithm could definitely be improved, but it would require a proper unit test due to the complexity of all possible situations. That seems to be too much work at the moment, and there is a chance that the behavior will change after we migrate to Subsquid (making the effort needles).